### ISSUE TITLE
com.cordovaplugincamerapreview.CameraActivity.onResume CameraActivity.java, line 297 java.lang.NullPointerException: Attempt to invoke interface method 'void com.cordovaplugincamerapreview.CameraActivity$CameraPreviewListener.onCameraStarted()' on a null object reference

### ISSUE THREAD

**Issue by ChanduCleqrQuote** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#650]
com.cordovaplugincamerapreview.CameraActivity.onResume CameraActivity.java:297
android.app.Fragment.performResume Fragment.java:2580
android.app.FragmentManagerImpl.moveToState FragmentManager.java:1346
android.app.FragmentManagerImpl.moveFragmentToExpectedState FragmentManager.java:1586
android.app.FragmentManagerImpl.moveToState FragmentManager.java:1647
android.app.FragmentManagerImpl.dispatchMoveToState FragmentManager.java:3057
android.app.FragmentManagerImpl.dispatchResume FragmentManager.java:3019
android.app.FragmentController.dispatchResume FragmentController.java:206
android.app.Activity.performResume Activity.java:8090
android.app.ActivityThread.performResumeActivity ActivityThread.java:4260
android.app.ActivityThread.handleResumeActivity ActivityThread.java:4302
android.app.servertransaction.ResumeActivityItem.execute ResumeActivityItem.java:52
android.app.servertransaction.TransactionExecutor.executeLifecycleState TransactionExecutor.java:176
android.app.servertransaction.TransactionExecutor.execute TransactionExecutor.java:97
android.app.ActivityThread$H.handleMessage ActivityThread.java:2044
android.os.Handler.dispatchMessage Handler.java:107
android.os.Looper.loop Looper.java:223
android.app.ActivityThread.main ActivityThread.java:7562
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:539
com.android.internal.os.ZygoteInit.main ZygoteInit.java:950

**Comment by ChanduCleqrQuote** [comment]
Anyone please let me know any work around for it. It is very urgent.

**Comment by ityoung2016** [comment]
I am having a similar problem on my cordova-built app for Android. It used to work but somebody changed something in the environment used by cordova and/or Android. The plugin still works perfectly in the cordova/Xcode environment. Help is needed!

**Comment by ChanduCleqrQuote** [comment]
Yes, in cordova/ios build it is working perfect. Only facing in android on some devices.

**Comment by ityoung2016** [comment]
I just checked three versions of the plugin (0.12.2, 0.12.1, 0.12.0) and all three failed on my Acer Android v7 but worked on my 2017 iPad.

**Comment by ChanduCleqrQuote** [comment]
Actually i am using 0.12.1, what are the changes came up in 0.12.2 ?

**Comment by ityoung2016** [comment]
You can access 0.12.2 through GitHub and then read the ChangeLog.

**Comment by ityoung2016** [comment]
I solved my problem but I do not know if I have solved yours. I checked the AndroidManifest.xml file at ~/platforms/android/app/src/main/AndroidManifest.xml and found that it was missing the permission for the camera. What I found was:

`    <uses-permission android:name="android.permission.RECORD_AUDIO" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
`
So I changed it to:

`   <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
`
I then used cordova to make and install a new version:

`   cordova run android --1234567890`

where the number is my Acer tablet SNID.

The app loaded and, when I chose the section where the camera is used, it asked for permission to use the camera. It had not done this during the many attemots I had made earlier to use the camera. I gave permission and the problem was solved.

The camera permission should have automatically been added, I thought, by the following in the config.xml file:

`    <edit-config file="AndroidManifest.xml" mode="merge" target="/manifest">
        <uses-permission android:name="android.permission.RECORD_AUDIO" />
        <uses-permission android:name="android.permission.CAMERA" />
        <uses-feature android:name="android.hardware.camera" />
    </edit-config>
`
but that clearly didn't help although the audio permission had been added as shown above.

I hope this helps.
